### PR TITLE
[realease/v0.5.0] Added entrypoint script

### PR DESCRIPTION
**Pull request by jonaphael**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Is there any issue related to this PR?** (Link to an issue, e.g. #99999)

* **Other information**:

### CODE REVIEW

**Review by eduardogmisiuk — DISMISSED** [review]
LGTM

File: subscription-engine/kafka-ws/bin/entryPoint.sh [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+if [ ! -z "${DEBUG+x}" ]; then
+    set -x
+fi
+
+export KAKKA_WS_CONNECTION_RETRY_COUNT=${KAFKA_WS_APP_CONNECTION_RETRY_COUNT:-"5"}
+export KAFKA_WS_CONNECTION_RETRY_TIMEOUT=${KAFKA_WS_CONNECTION_RETRY_TIMEOUT:-"3"}
+
+export KAFKA_WS_CONSUMER_METADATA_BROKER_LIST=${KAFKA_WS_CONSUMER_METADATA_BROKER_LIST:-"kafka:9092"}
+
+# Redis parameters
```
Comment (rascaraficci) [inline subscription-engine/kafka-ws/bin/entryPoint.sh:16]: I think it is better to put these settings close to the Redis' health check  code.

File: subscription-engine/kafka-ws/bin/entryPoint.sh [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+readonly REDIS_CONN_TIMEOUT=${REDIS_CONN_TIMEOUT:-"180"}
```
Comment (rascaraficci) [inline subscription-engine/kafka-ws/bin/entryPoint.sh:17]: You should keep the same patterns used to the Kafka settings.

File: subscription-engine/kafka-ws/bin/entryPoint.sh [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+echo -e "Connection established with **${address_splited[0]}** on port **${address_splited[1]}**\n"
+
+# REDIS CONNECTION CHECK
+
+# Waiting for redis for at most 3 minutes
+START_TIME=$(date +'%s')
+echo "Waiting for Redis fully start. Host '${REDIS_HOST}', '${REDIS_PORT}'..."
+echo "Try ping Redis... "
+PONG=$(redis-cli -h "${REDIS_HOST}" -p "${REDIS_PORT}" -a "${REDIS_PASSWD}" ping | grep PONG)
+while [ -z "${PONG}" ]; do
```
Comment (rascaraficci) [inline subscription-engine/kafka-ws/bin/entryPoint.sh:62]: Use the same logic used to the check the Kafka server based on the number of attempts.

File: subscription-engine/kafka-ws/bin/entryPoint.sh [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+# -e Exit imediatly if a comand exit with non-zero status
+# -x Print commands and their arguments as they are executed
+
+# Debug mode
+if [ ! -z "${DEBUG+x}" ]; then
+    set -x
+fi
+
+export KAKKA_WS_CONNECTION_RETRY_COUNT=${KAFKA_WS_APP_CONNECTION_RETRY_COUNT:-"5"}
```
Comment (rascaraficci) [inline subscription-engine/kafka-ws/bin/entryPoint.sh:11]: Why is it necessary to use `export` here?

**Review by rascaraficci — CHANGES_REQUESTED** [review]
To make easier the maintenance is better to use the same logic to check the connections with both  Kafka and Redis. In the proposed implementation, one is based on the number of attempts and the other on the elapsed time. It's also important to use the same pattern to the variables too.

File: subscription-engine/kafka-ws/bin/entryPoint.sh (hunk 4 above)
Comment (eduardogmisiuk) [reply subscription-engine/kafka-ws/bin/entryPoint.sh:11]: KAKKA -> KAFKA

Comment (thiagodpf) [reply subscription-engine/kafka-ws/bin/entryPoint.sh:11]: does it make sense for this variable to be called that? The others do not seem to have this same pattern:
> KAFKA_WS_**APP**_CONNECTION_RETRY_COUNT

File: subscription-engine/kafka-ws/bin/entryPoint.sh (hunk 2 above)
Comment (thiagodpf) [reply subscription-engine/kafka-ws/bin/entryPoint.sh:17]: Service variables did not have to start as: **KAFKA_WS_** REDIS_CONN_TIMEOUT ?

File: subscription-engine/kafka-ws/bin/entryPoint.sh (hunk 4 above)
Comment (jonaphael) [reply subscription-engine/kafka-ws/bin/entryPoint.sh:11]: > does it make sense for this variable to be called that?
> 
> > KAFKA_WS_**APP**_CONNECTION_RETRY_COUNT
> 
> The others do not seem to have this same pattern...

This pattern is the configManager pattern,m we don't want it to differ from development, I updated it.

File: subscription-engine/kafka-ws/bin/entryPoint.sh (hunk 2 above)
Comment (jonaphael) [reply subscription-engine/kafka-ws/bin/entryPoint.sh:17]: > Service variables did not have to start as: **KAFKA_WS_** REDIS_CONN_TIMEOUT ?

__note that__ This is the release pull request, the local variables can start with KAFKA_WS prefix, but not the developement will be.

File: subscription-engine/kafka-ws/bin/entryPoint.sh [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+export KAFKA_WS_CONNECTION_RETRY_TIMEOUT=${KAFKA_WS_CONNECTION_RETRY_TIMEOUT:-"3"}
+
+export KAFKA_WS_CONSUMER_METADATA_BROKER_LIST=${KAFKA_WS_CONSUMER_METADATA_BROKER_LIST:-"kafka:9092"}
+
+# Redis parameters
+readonly REDIS_CONN_TIMEOUT=${REDIS_CONN_TIMEOUT:-"180"}
+readonly REDIS_HOST=${REDIS_HOST:-"redis"}
+readonly REDIS_PORT=${REDIS_PORT:-"6379"}
+readonly REDIS_PASSWD=${REDIS_PASSWD:-""}
```
Comment (thiagodpf) [inline subscription-engine/kafka-ws/bin/entryPoint.sh:20]: Shouldn't these variables follow the same pattern as those in branch `development`?

This is the documentation for branch `development`:

> **Redis**
> Redis database management configuration
> 
> | Key | Purpose   | Default Value   | Valid Values | Environment variable
> ------| ----------| ----------------|--------------|------------------------
> redis.host| Redis host | kafka-ws-redis | string | KAFKA_WS_REDIS_HOST
> redis.port | Redis port | 6379 | integer | KAFKA_WS_REDIS_PORT
> redis.db | Redis database | 1 | integer | KAFKA_WS_REDIS_DB

Comment (eduardogmisiuk) [reply subscription-engine/kafka-ws/bin/entryPoint.sh:20]: By now, we shouldn't change this, the variable names will be changed in the next release.

**Review by thiagodpf — APPROVED** [review]

File: subscription-engine/kafka-ws/bin/entryPoint.sh [hunk 6]
```diff
@@ -8,22 +8,18 @@ if [ ! -z "${DEBUG+x}" ]; then
...
-export KAFKA_WS_CONSUMER_METADATA_BROKER_LIST=${KAFKA_WS_CONSUMER_METADATA_BROKER_LIST:-"kafka:9092"}
 
-# Redis parameters
-readonly REDIS_CONN_TIMEOUT=${REDIS_CONN_TIMEOUT:-"180"}
-readonly REDIS_HOST=${REDIS_HOST:-"redis"}
-readonly REDIS_PORT=${REDIS_PORT:-"6379"}
-readonly REDIS_PASSWD=${REDIS_PASSWD:-""}
+readonly KAFKA_WS_CONSUMER_METADATA_BROKER_LIST=${KAFKA_WS_CONSUMER_METADATA_BROKER_LIST:-"kafka:9092"}
 
 has_responded=false
```
Comment (rascaraficci) [inline subscription-engine/kafka-ws/bin/entryPoint.sh:18]: why not kafka_has_responded as redis_has_responded?

**Review by rascaraficci — APPROVED** [review]
It looks good to me!

**Review by eduardogmisiuk — APPROVED** [review]

**Pull request merged**